components/speedreader/resources/speedreader-desktop.js

```diff
@@ -147,17 +152,72 @@ const extractTextToSpeak = () => {
     }
 }
 
+const highlightWord = (paragraph, start, end) => {
+    nodes = document.createNodeIterator(
+        paragraph,
+        NodeFilter.SHOW_TEXT,
+        { acceptNode(n) { return NodeFilter.FILTER_ACCEPT } })
+
+    const range = new Range()
+
+    let startNode = null
+    let endNode = null
+
+    while (node = nodes.nextNode()) {
+        if (!startNode) {
+            if (start < node.textContent.length) {
+                startNode = node
+                range.setStart(node, start)
+            }
+            start -= node.textContent.length
+        }
+        if (!endNode) {
+            if (end <= node.textContent.length) {
+                endNode = node
+                range.setEnd(endNode, end)
+            }
+            end -= node.textContent.length
+        }
+    }
+
+    CSS.highlights.set('tts-highlighted-word', new Highlight(range))
+}
+
+const setTtsButtonIcon = (e, i) => {
+    const button = e?.querySelector('.tts-paragraph-player-button')
+    if (button) {
+        button.classList.remove('tts-play-icon', 'tts-pause-icon')
+        button.classList.add(i)
+    }
+}
+
 const highlightText = (ttsParagraphIndex, charIndex, length) => {
     document.querySelectorAll('.tts-highlighted').forEach((e) => {
         if (e.getAttribute('tts-paragraph-index') != ttsParagraphIndex) {
             e.classList.remove('tts-highlighted')
+            setTtsButtonIcon(e, 'tts-play-icon')
         }
     })
 
     const paragraph = document.querySelector(
         '[tts-paragraph-index="' + ttsParagraphIndex + '"]')
     if (paragraph) {
         paragraph.classList.add('tts-highlighted')
+        if (window.speedreaderData.ttsReading) {
+            setTtsButtonIcon(paragraph, 'tts-pause-icon')
+        } else {
+            setTtsButtonIcon(paragraph, 'tts-play-icon')
+        }
+        highlightWord(paragraph, charIndex, charIndex + length)
+    } else {
+        CSS.highlights.clear()
+    }
+}
+
+const setTtsReadingState = (reading) => {
```

Comment:
as mentioned above we should namespace any globals for clarity and also to avoid conflicts with other brave scripts running in the isolated world